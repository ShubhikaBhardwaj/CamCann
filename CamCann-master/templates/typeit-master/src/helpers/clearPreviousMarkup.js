/**
 * Given an element, remove any markup left from a previous instance of TypeIt.
 */
export default (element, isInput = false) => {
  if (isInput) {
    element.value = "";
    return;
  }

  if (element.querySelector(".ti-cursor")) {
    element.innerHTML = "";
  }
};
