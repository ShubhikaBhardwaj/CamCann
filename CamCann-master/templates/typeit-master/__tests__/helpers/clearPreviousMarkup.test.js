import clearPreviousMarkup from "../../src/helpers/clearPreviousMarkup";

test("Should remove .ti-* markup.", () => {
  setHTML`<h3 id="example1" data-typeitid="arbitrary-id">hello!<span class="ti-cursor">|</span></h3>`;

  clearPreviousMarkup(document.getElementById("example1"));

  expect(document.getElementById("example1").outerHTML).toMatchSnapshot();
});

test("Should leave other markup alone.", () => {
  setHTML`<h3 id="example1" data-typeitid="arbitrary-id">This should be left alone.</h3>`;

  clearPreviousMarkup(document.getElementById("example1"));

  expect(document.getElementById("example1").outerHTML).toMatchSnapshot();
});
