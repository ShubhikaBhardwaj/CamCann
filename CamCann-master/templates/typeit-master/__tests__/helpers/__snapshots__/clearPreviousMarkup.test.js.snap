// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should leave other markup alone. 1`] = `"<h3 id=\\"example1\\" data-typeitid=\\"arbitrary-id\\">This should be left alone.</h3>"`;

exports[`Should remove .ti-* markup. 1`] = `"<h3 id=\\"example1\\" data-typeitid=\\"arbitrary-id\\"></h3>"`;
